rosys/persistence.py

```diff
@@ -28,7 +32,7 @@ def backup(self):
             json.dump(self.dump(), f)
 
     def parse_world(self, obj: dict):
-        return World.parse_obj(obj)
+        return self.world.parse_obj(obj)  # NOTE we use the provided world for parsing to load all attributes
```

Comment:
I guess the purpose of this change is that `parse_world` will return whatever type `self.world` is.
As far as I understand the instance of `self.world` is omitted, because `parse_obj` is a class method.
`parse_world` is missing a return type. Since it should return "what `self.world` is", we can use a [TypeVar](https://docs.python.org/3/library/typing.html#typing.TypeVar): Introduce `W = TypeVar('W')` and use it as `self.world: W` and `def parse_world(...) -> W:`.